apps/server/src/modules/authorization/domain/rules/board-do.rule.ts

```diff
@@ -60,14 +65,22 @@ export class BoardDoRule implements Rule {
 	}
 
 	private hasPermissionForDrawingElement(userWithBoardRoles: UserWithBoardRoles): boolean {
-		// TODO
-		// for now, board Readers have also write permission for DrawingElement.
-		// This is needed because files "attached" to drawing element ask permission for the drawing element itself
-		// this will be changed in the future, when there should be a separate permission for drawing element and
-		// separate permission for files attached to drawing element (when drawing element is parent)
 		return this.isBoardReader(userWithBoardRoles);
 	}
 
+	private shouldProcessDrawingElementFile(
+		boardDoAuthorizable: BoardDoAuthorizable,
+		context: AuthorizationContext
+	): boolean {
+		return (
+			isDrawingElement(boardDoAuthorizable.boardDo) && context.requiredPermissions.includes(Permission.FILESTORAGE_VIEW)
```

Comment:
so, if the function call does not contain required permission FILESTORAGE_VIEW, you simply do not process this.
Is interesting, but not so clear this logic. At the very least, write in a comment why